TFlippy_TerritoryControl_Core_Dev/Entities/Materials/MinableMats.as

```diff
@@ -0,0 +1,120 @@
+#define SERVER_ONLY
+
+#include "MakeMat.as";
+#include "Hitters.as";
+/*
+NOTE: This must be put before Generic Hit as generic hit sets the damage to 0 and AFTER all damage resistance effects such as WoodHit.as
+										 MinableMats.as;
+*/
+class HarvestBlobMat
+{
+	f32 amount;
+	string matname;
+	HarvestBlobMat(f32 pamount, string pmatname)
+	{
+		amount = pamount;
+		matname = pmatname;
+	}
+};
+
+void onInit(CBlob@ this)
```

Comment:
What's the point of putting it there?
Why can't you remove all that stuff completely and just assign an empty array for material yield, and just assign actual arrays in onInit() of blobs that use this script?

Reply:
The reason why i dont have this set in each blob is because the previous method also used a large file and this gives a better overview for adding all the new things i needed to add. Obviously this can be put in the init of the blob instead if you want.
Ill put the object definition into a common file for easier use for that.

Additionaly many blobs dont have unique innitiations so i would have to move the split into for example door.as

Reply:
The problem with large files (other than the fact that each time you create 'triplanewreck' it has to compare name to 36 other strings (for now, might be much more (maybe for other blob) later)) is each time you need to add a new thing you need to change that large file. Might not sound bad with just one file, but imagine if you had to do that for every single script you include into a blob.
Same stands for deleting old blobs. If you delete a blob forget to remove an "if" from that file, it'll compare blob name extra time for literally no reason and there won't be an easy way to notice it.

Reply:
Also a million "if"s instead of a single switch or use of dictionary/hash map/whatever is just not a good sign.

Reply:
I know as i said it was mostly a temporary method due to copying the old file and wanting to set everything up, for example all the wrecks.